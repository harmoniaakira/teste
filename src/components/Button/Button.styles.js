import styled from 'styled-components'

export const ButtonComponent = styled.button`
  width: 280px;
  min-height: 48px;
  font-size: 12px;
  line-height: 24px;
  font-weight: 700;
  border: none;
  border-radius: 100px;
  box-shadow: 0px 6px 10px rgba(0, 0, 0, 0.11);
  transition: background-color 0.5s ease;
  text-transform: uppercase;
  cursor: pointer;

  ${props => props.fullScreen 
    ? "margin: 0 auto;"
    : ""
  }

  ${props => props.disabled 
    ? "background-color: #E0E0E0;" +
      "color: #8E8E93;" +
      "cursor: default;"
    : "background-color: #054375;" +
      "color: white;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: #E0E0E0;"
      : "background: #043C64;"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`
export const ButtonPopup = styled.button`
  width: 280px;
  height: 48px;
  font-size: 14px;
  line-height: 16px;
  font-weight: 600;
  letter-spacing: 0.75px;
  border: none;
  border-radius: 100px;
  text-transform: uppercase;
  transition: background-color 0.5s ease;
  cursor: pointer;
  background-color: transparent;

  ${props => props.disabled 
    ? "color: #8E8E93;" +
      "cursor: default;"
    : "color: #054375;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: transparent;"
      : "background: rgba(5, 67, 117, .1);"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`