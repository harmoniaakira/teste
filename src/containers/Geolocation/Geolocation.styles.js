import styled from 'styled-components'

export const GeoContainer = styled.div`
  @media (max-width: 1439px) {
    margin: 0 auto;
    display: flex;
    flex-flow: column;
    height: 100%;
    text-align: center;
  }
`

export const GeoMobileView = styled.div`
  height: 100%;
  width: 100%;
  max-width: 768px;
  
  margin: 0 auto;

  display: flex;
  flex-direction: column;

  background-color: #fff;

  @media (min-width: 1440px) {
    display: none;
  }
`

export const Flexgrow = styled.div`
  flex: 1 1 auto;
`

export const MobileCheck = styled.div`
  display: flex;
  gap: 5px;
  max-width: 318px;
  margin: auto;

  @media (min-width: 1440px) {
    gap: 0;
    margin: 0;
    max-width: 380px;
  }
`

export const MobileCheckLabel = styled.div`
  color: #636366;
  font-size: 16px;
  font-weight: 700;
  line-height: 24px;
  letter-spacing: 0.2px;
`

export const Content = styled.div`
  width: 518px;
  min-height: 170px;
  max-height: 928px;
  margin: 0;
  position: absolute;
  top: 50%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);

  @media (min-width: 1440px) {
    width: unset;
  }
`

export const Info = styled.div`
  display: flex;
`

export const Avatar = styled.img`
  margin-right: 50px;
`

export const Title = styled.h2`
  color: #222222;
  font-size: 20px;
  font-weight: 900;
  margin-top: 48px;
  line-height: 28px;
`

export const Text = styled.p`
  color: #636363;
  max-width: 318px;
  margin: ${props => props.centered ? "0 auto" : "0"};
  letter-spacing: 0.2px;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  margin-bottom: 30px;

  @media (min-width: 1440px) {
    max-width: 368px;
  }
`

export const FootNote = styled.p`
  color: #222222;
  letter-spacing: 0.2px;
  font-size: 12px;
  font-weight: 600;
  line-height: 46px;
  margin-bottom: 80px;
  width: 308px;
  margin: auto;
  text-align: left;
  margin-bottom: 80px;

  @media (min-width: 1440px) {
    margin: 0 0 90px 0;
  }
`