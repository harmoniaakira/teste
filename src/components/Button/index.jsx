import React from 'react'
import PropTypes from 'prop-types';
import { ButtonComponent, ButtonPopup } from './Button.styles'

const Button = ( { handleClick, disabled, children, popup } ) => {
    
	return ( popup 
		? 	<ButtonPopup 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonPopup>
		:	<ButtonComponent 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonComponent>
	)
}

Button.propTypes = {
	handleClick: PropTypes.func.isRequired,
	children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired,
	disabled: PropTypes.bool,
	popup: PropTypes.bool
	
}

Button.defaultProps = {
	disabled: false,
	popup: false
}

export default Button
