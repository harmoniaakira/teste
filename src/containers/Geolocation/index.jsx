import { MobileView } from 'components/Layout/Layout.styles'
import { StyledCheckbox } from 'components/StyledCheckbox/StyledCheckbox.styles'
import Web from 'components/Web'
import React, { useState } from 'react'
import { useHistory } from "react-router-dom"
import avatar from '../../assets/images/GeolocationView/avatar-geolocation.svg'
import AppHeader from '../../components/AppHeader'
import Button from '../../components/Button'
import Dialog from '../../components/Dialog'
import {
  Avatar, Content, Flexgrow, FootNote, GeoContainer, Info, MobileCheck, MobileCheckLabel, Text, Title
} from './Geolocation.styles'


const userGeolocationInitState = {
  coords: {
    longitude: 0,
    latitude: 0,
    altitude: 0
  },
  timestamp: 0
}

const Geolocation = () => {
  const history = useHistory();
  const [isChecked, setIsChecked] = useState(false);
  const [open, setOpen] = useState(false);
  const [userGeolocation, setUserGeolocation] = useState(userGeolocationInitState);

  const getCurrentPositionSuccess = (pos) => {
    setUserGeolocation(pos)
    console.log('userGeolocation: ', userGeolocation)
  }

  const getCurrentPosition = () => {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(getCurrentPositionSuccess);
    }
  }

  const handleClose = () => {
    setOpen(false)
  }

  const submitGeolocation = () => {
    setOpen(false);
    getCurrentPosition();
    console.log('submiting Geolocation');
    history.push('/onboarding-pf-frontend/cadastro');
  }

  const handleClick = () => {
    if (isChecked) {
      submitGeolocation();
    } else {
      setOpen(true);
    }
  }

  const handleToggle = () => {
    setIsChecked(!isChecked);
  }

  return (
    <GeoContainer>
      <MobileView $padding="0 0 40px 0">
        <AppHeader avatar={avatar} />
        <Flexgrow>
          <Title>Compartilhe sua localização</Title>
          <Text centered>
            Para prosseguirmos com a proposta, precisamos da sua permissão. :)
          </Text>

          <MobileCheck>
            <MobileCheckLabel> Autorizo compartilhar minha localização e dados do aparelho. </MobileCheckLabel>
            <StyledCheckbox
              value={isChecked}
              onChange={handleToggle}
            />
          </MobileCheck>
          <FootNote>É necessário aceitar somente uma vez.</FootNote>

          <Button handleClick={handleClick}>Continuar</Button>
        </Flexgrow>
      </MobileView>

      <Web>
          <Content>
            <Info>
              <Avatar src={avatar} />
              <div>
                <Title>Compartilhe sua localização</Title>
                <Text>
                  Para prosseguirmos com a proposta, precisamos da sua permissão. :)
                </Text>
                <MobileCheck>
                  <MobileCheckLabel> Autorizo compartilhar minha localização e dados do aparelho. </MobileCheckLabel>
                  <StyledCheckbox
                    value={isChecked}
                    onChange={handleToggle}
                  />
                </MobileCheck>
                <FootNote>É necessário aceitar somente uma vez.</FootNote>
              </div>
            </Info>
            <Button handleClick={handleClick}>Continuar</Button>
          </Content>
      </Web>

      <Dialog
        open={open}
        title="Atenção"
        confirmButtonLabel="continuar"
        cancelButtonLabel="voltar"
        onConfirm={submitGeolocation}
        onCancel={handleClose}
        handleClose={handleClose}
      >
        Sem aceitar compartilhar a localização, a solicitação estará sujeita a análise
        diferenciada da proposta. Deseja seguir sem compartilhar a localização?
      </Dialog>
    </GeoContainer >
  )
}

export default Geolocation;
