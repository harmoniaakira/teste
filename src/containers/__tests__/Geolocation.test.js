import React from 'react';
import { shallow } from 'enzyme';
import Button from 'components/Button';
import Geolocation from '../Geolocation'


describe('Geolocation', () => {

    const geolocation = shallow(<Geolocation />)
    
    it("should run onClick function", () => {
        geolocation.find(Button).at(0).props().handleClick();
    });

    it('should render Geolocation page', () => {
        expect(geolocation).toMatchSnapshot()
    })

})