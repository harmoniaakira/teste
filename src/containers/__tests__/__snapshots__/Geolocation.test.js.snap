// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geolocation should render Geolocation page 1`] = `
<styled.div>
  <styled.div
    $padding="0 0 40px 0"
  >
    <AppHeader
      avatar="SvgrURL"
    />
    <styled.div>
      <styled.h2>
        Compartilhe sua localização
      </styled.h2>
      <styled.p
        centered={true}
      >
        Para prosseguirmos com a proposta, precisamos da sua permissão. :)
      </styled.p>
      <styled.div>
        <styled.div>
           Autorizo compartilhar minha localização e dados do aparelho. 
        </styled.div>
        <Styled(WithStyles(ForwardRef(Checkbox)))
          onChange={[Function]}
          value={false}
        />
      </styled.div>
      <styled.p>
        É necessário aceitar somente uma vez.
      </styled.p>
      <Button
        disabled={false}
        handleClick={[Function]}
        popup={false}
      >
        Continuar
      </Button>
    </styled.div>
  </styled.div>
  <Web>
    <styled.div>
      <styled.div>
        <styled.img
          src="SvgrURL"
        />
        <div>
          <styled.h2>
            Compartilhe sua localização
          </styled.h2>
          <styled.p>
            Para prosseguirmos com a proposta, precisamos da sua permissão. :)
          </styled.p>
          <styled.div>
            <styled.div>
               Autorizo compartilhar minha localização e dados do aparelho. 
            </styled.div>
            <Styled(WithStyles(ForwardRef(Checkbox)))
              onChange={[Function]}
              value={false}
            />
          </styled.div>
          <styled.p>
            É necessário aceitar somente uma vez.
          </styled.p>
        </div>
      </styled.div>
      <Button
        disabled={false}
        handleClick={[Function]}
        popup={false}
      >
        Continuar
      </Button>
    </styled.div>
  </Web>
  <Memo(DialogSlide)
    cancelButtonLabel="voltar"
    confirmButtonLabel="continuar"
    handleClose={[Function]}
    onCancel={[Function]}
    onConfirm={[Function]}
    open={true}
    title="Atenção"
  >
    Sem aceitar compartilhar a localização, a solicitação estará sujeita a análise diferenciada da proposta. Deseja seguir sem compartilhar a localização?
  </Memo(DialogSlide)>
</styled.div>
`;
